File: A2_MODULO3DanielRomero/app/src/main/java/com/example/actividad2modulo3danielromero/MainActivity.java
package com.example.actividad2modulo3danielromero;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Button;
import android.widget.TextView;
import android.util.Log;

public class MainActivity extends AppCompatActivity {
    TextView total;
    Button sum, mult, div, resta;
    EditText n1, n2;
    float n1_float, n2_float, total_float = 0;
    private static final String TAG = "Calculadora Simple";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sum = (Button) findViewById(R.id.sum);
        mult = (Button) findViewById(R.id.mult);
        resta = (Button) findViewById(R.id.resta);
        div = (Button) findViewById(R.id.div);

        n1 = (EditText) findViewById(R.id.n1);
        n2 = (EditText) findViewById(R.id.n2);
        total = (TextView) findViewById(R.id.total);

        Log.d(TAG, " INICIO APP CALCULADORA");
//        System.out.println("----------------------------------------------------");


        sum.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Log.d(TAG, " INICIO SUMA");

                System.out.println(n1.getText());
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float + n2_float;

                total.setText(total_float+"");
                Log.d(TAG, " FIN SUMA, TOTAL: "+String.valueOf(total_float)+"");

            }
        });

        resta.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Log.d(TAG, " INICIO RESTA");
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float - n2_float;

                total.setText(total_float+"");
                Log.d(TAG, " FIN RESTA, TOTAL: "+String.valueOf(total_float)+"");

            }
        });

        mult.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Log.d(TAG, " INICIO MULTIPLICACION");
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float * n2_float;

                total.setText(total_float+"");
                Log.d(TAG, " FIN MULTIPLICACION, TOTAL: "+String.valueOf(total_float)+"");

            }
        });

        div.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                Log.d(TAG, " INICIO DIVISION");
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 1;
                }

                total_float = n1_float/n2_float;

                total.setText(total_float+"");
                Log.d(TAG, " FIN DIVISION, TOTAL: "+String.valueOf(total_float)+"");

            }
        });
    }
}

File: A5_MODULO4_DANIELROMERO/app/src/main/java/com/root/a5_modulo4_danielromero/MainActivity.kt
package com.root.a5_modulo4_danielromero

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.DecimalFormat

class MainActivity : AppCompatActivity() {

    private val LOG_TAG = MainActivity::class.java.simpleName
    var total_eur: TextView? = null
    var convertir_to_eur: Button? = null
    var switch_converter_to_CLP: Button? = null
    var clp_input: EditText? = null
    var clp_float = 0.0
    var total_float_eur: Double = 0.0
    val EXTRA_MESSAGE = "com.example.a3_modulo3_daniel_romero.extra.MESSAGE"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        convertir_to_eur = findViewById<View>(R.id.convertir_to_eur) as Button
        clp_input = findViewById<View>(R.id.clp_input) as EditText
        total_eur = findViewById<View>(R.id.total_eur) as TextView
        switch_converter_to_CLP = findViewById<View>(R.id.switch_converter_btn) as Button

        convertir_to_eur!!.setOnClickListener {
            if (!clp_input!!.text.toString().trim { it <= ' ' }.isEmpty()) {
                clp_float = (clp_input!!.text.toString() + "").toFloat().toDouble()
            } else {
                clp_float = 1.0
                clp_input!!.setText("1")
            }
            total_float_eur = clp_float / 952.57
            val df = DecimalFormat("0.000")
            total_eur!!.text = df.format(total_float_eur) + ""
        }

    }

    fun switchDivisa(view: View?) {
        val intent = Intent(this, eur_to_clp::class.java)
        val message = clp_input!!.text.toString()
        intent.putExtra(EXTRA_MESSAGE, message)
        startActivity(intent)
        Log.d(LOG_TAG, "Cambio de Convertidor a EUROS")
    }

}